zhaquirks/ikea/__init__.py

```diff
@@ -73,6 +75,70 @@ class ScenesCluster(CustomCluster, Scenes):
     )
 
 
+class LevelControlCluster(CustomCluster, LevelControl):
+    """Ikea LevelControl Cluster for SYMFONISK sound remote Gen2."""
+
+    cluster_id = LevelControl.cluster_id
+
+    def __init__(self, *args, **kwargs):
+        """Init."""
+        super().__init__(*args, **kwargs)
+        self.endpoint.device.levelcontrol_bus.add_listener(self)
+
+
+class V1ShortcutCluster(CustomCluster):
+    """Ikea Shortcut Button cluster."""
+
+    name = "V1ShortcutCluster"
+    cluster_id = 0xFC7F
+
+    server_commands = {
+        0x01: foundation.ZCLCommandDef(
+            "shortcut",
+            {
+                "shortcut_mode": t.int8s,
+                "param1": t.int8s,
+            },
+            False,
+            is_manufacturer_specific=True,
+        ),
+    }
+
+    def handle_cluster_request(
+        self,
+        hdr: foundation.ZCLHeader,
+        args: List[Any],
+        *,
+        dst_addressing: Optional[
+            Union[t.Addressing.Group, t.Addressing.IEEE, t.Addressing.NWK]
+        ] = None,
+    ):
+        """Handle a cluster request."""
+        self.debug(
+            "%s: handle_cluster_request - Command: %s Data: %s",
+            self.name,
+            hdr.command_id,
+            args,
+        )
+
+        if hdr.command_id == 0x01:
+            self.debug("%s: sending ZHA event", self.name)
+
+            event_args = {
+                "step_mode": (
+                    LevelControl.StepMode.Up
+                    if args.shortcut_mode == 1
+                    else LevelControl.StepMode.Down
+                ),
+                "step_size": args.param1,
+                "transition_time": 0,
+            }
+
+            self.endpoint.device.levelcontrol_bus.listener_event(
+                "listener_event", ZHA_SEND_EVENT, "step_with_on_off", event_args
+            )
```

Comment:
Could you try to replace with:
```python
            self.endpoint.level.command(
                0x06, event_args, expect_reply=True
            )
```

If we make this work, all the `levelcontrol_bus` can be removed.

Reply:
I've experimented a bit with that. While ```self.endpoint.level``` does not exist, I've ended up with
```
self.endpoint.out_clusters[8].command(
    0x06, event_args, expect_reply=True
)
```

Sadly, this did not fire zha_events.

Reply:
Just calling the following on the ``LevelControl`` cluster might work (not on a bus, directly on the the cluster):
```python
listener_event(ZHA_SEND_EVENT, "step_with_on_off", event_args)
```

Reply:
@TheJulianJES Thanks! That worked :-)